### PR TITLE
IDENTITY-5971 :  Fix Userinfo subject format inconsistency among grant types

### PR DESCRIPTION
**Pull request by DMHP**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 1]
```diff
@@ -102,6 +111,11 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         if(claims == null){
             claims = new HashMap<String,Object>();
         }
+        for (Map.Entry<String, Object> entry : claims.entrySet()) {
+            if(entry.getKey().equals("sub")){
```
Comment (duchuyctlk) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: `"sub".equals(entry.getKey())` could help avoid NPE in the case `entry.getKey()` returns a null value

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 2]
```diff
@@ -102,6 +111,11 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         if(claims == null){
             claims = new HashMap<String,Object>();
         }
+        for (Map.Entry<String, Object> entry : claims.entrySet()) {
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Use subject claim directly  from the map

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 3]
```diff
@@ -214,4 +228,58 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
+     * Returns subject claim.
+     *
+     * @param subject subject
+     * @param tenantDomain tenant domain
+     * @param tokenResponse token response
+     * @return
+     * @throws UserInfoEndpointException
+     */
+
+    private String returnSubjectClaim(Object subject, String tenantDomain, OAuth2TokenValidationResponseDTO tokenResponse)
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Remove Object type

Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: make protected

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 4]
```diff
@@ -214,4 +228,58 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
+            throws UserInfoEndpointException {
+        String clientId;
+        String sub = null;
+        try {
+            clientId = OAuth2Util.getClientIdForAccessToken
+                    (tokenResponse.getAuthorizationContextToken().getTokenString());
+        } catch (IdentityOAuth2Exception e) {
+            throw new UserInfoEndpointException("Error while obtaining service provider access token clientID", e);
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: make the error descriptive

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 5]
```diff
@@ -214,4 +228,55 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
+     * @param sub subject
+     * @param tenantDomain tenant domain
+     * @param tokenResponse token response
+     * @return
+     * @throws UserInfoEndpointException
+     */
+
+    protected String returnSubjectClaim(String sub, String tenantDomain, OAuth2TokenValidationResponseDTO tokenResponse)
+            throws UserInfoEndpointException {
+        String clientId;
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: add a new line

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 6]
```diff
@@ -214,4 +228,55 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
+            //getting service provider
+            serviceProvider = applicationMgtService.getServiceProviderByClientId(
+                    clientId, IdentityApplicationConstants.OAuth2.NAME, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            throw new UserInfoEndpointException("Error while obtaining the service provider.", e);
+        }
+        String userName = tokenResponse.getAuthorizedUser();
+        String userStoreDomain = IdentityUtil.extractDomainFromName(userName);
+
+
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: remove one line

File: pom.xml [hunk 7]
```diff
@@ -540,7 +540,7 @@
         <carbon.kernel.registry.imp.pkg.version.range>[1.0.1, 2.0.0)</carbon.kernel.registry.imp.pkg.version.range>
 
         <!-- Carbon Identity Framework version -->
-        <carbon.identity.framework.version>5.7.0</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.8.0-SNAPSHOT</carbon.identity.framework.version>
```
Comment (pulasthi7) [inline pom.xml]: This should be a released version. Snapshot versions are removed when the next version is released.

**Pull request merged**